packages/server/lib/browsers/electron.ts

```diff
@@ -347,8 +347,14 @@ export = {
         url: downloadItem.getURL(),
       })
 
-      downloadItem.once('done', () => {
-        automation.push('complete:download', {
+      downloadItem.once('done', (_event, state) => {
+        if (state === 'completed') {
+          automation.push('complete:download', {
+            id: downloadItem.getETag(),
+          })
+        }
+
+        automation.push('canceled:download', {
```

Comment:
If state is completed, should it push both 'complete:download' _and_ 'canceled:download'?

Reply:
I dont think so - the browser reports these separately. I pulled the canceled state from CDP's states: https://chromedevtools.github.io/devtools-protocol/tot/Browser/#event-downloadProgress

Thoughts on why pushing both?

Reply:
Isn't that what it's currently doing? It doesn't return early, so it's going to push twice if state is completed.

Reply:
Oh I see what you mean.  yes it should return fast